### DISCUSSION TITLE
Firefox Permanence?

### DISCUSSION THREAD

**Discussion by bigblackplastic** [EFForg/https-everywhere#20161] [Q&A] [answered]
Hi,

HTTPS Everywhere in Firefox writes to disk, even in private mode. Interactions with it constantly cause data to be written to storage-sync-v2.sqlite-wal.

What exactly is being written here?

**Comment by pseacrest** [comment]
can you share details of what you are seeing being stored there?

**Reply by bigblackplastic** [reply]
No. If I could I wouldn't have asked about it here.

The data is saved to a .sqlite-wal file, which I cannot read. Normally I would simply read the main .sqlite database once it has been updated, but this never happens here: the .sqlite-wal file is never deleted, and the main database is never updated. Hence, I cannot see what is being written to disk.

**Comment by zoracon** [comment, marked as answer]
For [context](https://blog.mozilla.org/addons/2020/07/09/changes-to-storage-sync-in-firefox-79/), extensions can save data either in local storage that is limited to the current profile. Or sync storage, which is synchronized across your Firefox installations if you have Firefox Sync turned on. 

A .sqlite-wal extension means "write ahead log" file which the SQLite engine utilizes to write at a faster pace to recover data.

You should be able to check extension storage and what it entails under `about:devtools-toolbox?id=https-everywhere%40eff.org&type=extension`.
